tests/devices/test_default_mixed_jax.py

```diff
@@ -0,0 +1,635 @@
+
+        # Check that the correct differentiation method is being used.
+        if diff_method == "backprop":
+            assert circuit.gradient_fn == "backprop"
+        elif diff_method == "parameter-shift":
+            assert circuit.gradient_fn is qml.gradients.param_shift
+        else:
+            assert circuit.gradient_fn is qml.gradients.finite_diff
+
+        res = jax.grad(cost)(params)
+
+        expected_grad = (
+            np.array(
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res, expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.xfail(reason="Line 230 in QubitDevice: results = self._asarray(results) fails")
+    @pytest.mark.parametrize(
+        "dev_name,diff_method,mode",
+        [
+            ["default.mixed", "finite-diff", "backward"],
+            ["default.mixed", "parameter-shift", "backward"],
+            ["default.mixed", "backprop", "forward"],
+        ],
+    )
+    def test_ragged_differentiation(self, dev_name, diff_method, mode, tol):
+        """Tests correct output shape and evaluation for a tape
+        with prob and expval outputs"""
+
+        dev = qml.device(dev_name, wires=2)
+        x = jnp.array(0.543)
+        y = jnp.array(-0.654)
+
+        @qml.qnode(dev, diff_method=diff_method, mode=mode, interface="jax")
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return [qml.expval(qml.PauliZ(0)), qml.probs(wires=[1])]
+
+        res = circuit(x, y)
+        expected = np.array(
+            [
+                np.cos(x),
+                (1 + np.cos(x) * np.cos(y)) / 2,
+                (1 - np.cos(x) * np.cos(y)) / 2,
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = jax.jacobian(circuit, (0, 1))(x, y)
+        expected = np.array(
+            [
+                [-np.sin(x), -np.sin(x) * np.cos(y) / 2, np.cos(y) * np.sin(x) / 2],
+                [0, -np.cos(x) * np.sin(y) / 2, np.cos(x) * np.sin(y) / 2],
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("jacobian_fn", [jax.jacfwd, jax.jacrev])
+    @pytest.mark.parametrize("decorator", decorators)
+    def test_batching(self, jacobian_fn, decorator, tol):
+        """Tests that the gradient of the qnode is correct with batching"""
+        dev = qml.device("default.mixed", wires=2)
+
+        if decorator == jax.jit:
+            pytest.skip("Batching currently not supported for JAX jit")
```

Comment:
Not sure why, but this is giving incorrect results. It also occurs for the `default.qubit` device (see https://github.com/PennyLaneAI/pennylane/issues/2762) so I assumed it was something to do with batching with jit.

Reply:
Got it. Best to add a TODO here to clear it up (and potentially making the skipping an xfail) and mention this line in the linked issue too. Skipping sounds a bit as though the issue was a feature yet to be added, instead of unexpected behaviour.